mutiny-core/src/ldkstorage.rs

```diff
@@ -88,26 +93,53 @@ impl<S: MutinyStorage> MutinyNodePersister<S> {
         format!("{}_{}", key, self.node_id)
     }
 
-    fn persist_local_storage<W: Writeable>(
+    fn init_persist_local_storage<W: Writeable>(
         &self,
-        key: &str,
+        key: String,
         object: &W,
-        version: Option<u32>,
-    ) -> Result<(), lightning::io::Error> {
-        let key_with_node = self.get_key(key);
-        self.storage
-            .set_data(key_with_node, object.encode(), version)
-            .map_err(|e| {
-                match e {
-                    MutinyError::PersistenceFailed { source } => {
-                        log_error!(self.logger, "Persistence failed on {key}: {source}");
+        version: u32,
+        update_id: MonitorUpdateIdentifier,
+    ) -> ChannelMonitorUpdateStatus {
+        let storage = self.storage.clone();
+        let completed_updates = self.completed_updates.clone();
+        let logger = self.logger.clone();
+        let object = object.encode();
+        spawn(async move {
+            let mut attempts = 0;
+            loop {
+                match persist_local_storage(&storage, &key, &object, Some(version), &logger).await {
+                    Ok(()) => {
+                        log_debug!(
+                            logger,
+                            "Persisted channel monitor: {:?}",
+                            update_id.funding_txo
+                        );
+
+                        // Sleep after monitor is persisted to give chance for the
+                        // manager to be persisted
+                        sleep(1_000).await;
+
+                        completed_updates.lock().await.push_back(update_id);
```

Comment:
We can't have the chain monitor in the persister because the chain monitor needs the persister, would be circular. 

The queue was the only way I could think to get around that.

Reply:
We still can't add to the persister retroactively because we need to have it in an `Arc` to give it to the `ChainMonitor`

Reply:
I guess if it was `Arc<Mutex<Option<Arc<ChainMonitor>>>>` we could